### ISSUE TITLE
Is react required? 

### ISSUE THREAD

**Issue by kevinkhill** [ts-engine/ts-engine#49]
I was under the assumption this could be a drop in for a working project. I tried to use this and the first command gave me

![Screenshot_2020-05-15-19-23-13](https://user-images.githubusercontent.com/266076/82108244-0cd0b080-96e2-11ea-8fd6-4a2b8ee36eff.png)

**Comment by kevinkhill** [comment]
I fixed the error by `yarn eslint --init` with typescript settings, but it still warns about react

**Comment by LeeCheneler** [comment]
Hey thanks for contributing!

If your project uses react then you'll want to use the `@ts-engine/babel-preset-react`. You can see how to configure it here: https://ts-engine.dev/docs/react-support.

If you're still seeing issues then let me know and I'll look into it some more!

**Comment by kevinkhill** [comment]
Hmm, I guess I should try and explain better. I have a TS project that uses eslint, prettier, and jest. It is just a library, so no react or a11y or the likes. I wanted to demo what using it would be like before taking out all the explicit testing/build dependencies

**Comment by LeeCheneler** [comment]
Ah OK cool, so the default linting config that is applied internally can be found here:

https://github.com/ts-engine/ts-engine/blob/master/packages/eslint-config/index.js

The default lint config does include react and a11y etc by default. We upgraded to ESLint 7 in release 1.5.0, perhaps our e2e tests have missed a regression. I'll run through the linting tests and check it out.

Have you customised the ESLint config at all? Or have any other ESLint dependencies that might be interfering or do you just have `@ts-engine/cli` as your only dev dependency?

**Comment by kevinkhill** [comment]
I haven't pushed anything, but this is the repo I am playing in

https://github.com/kevinkhill/ts-tokenizr

I wanted to see how it would work, and if easy, I will definitely start to use it more. I have never liked starting up a new TS project. All the configs and everything.

**Comment by kevinkhill** [comment]
Could the default be refactored into a base / react / recommended like others do, so I can opt out by using base, but others would still get default?

**Comment by LeeCheneler** [comment]
I've pulled your repo and just working through it now. This has been really useful thanks, I'm using ts-engine on quite a lot of packages now but nothing beats other people packages for hardening a tool!!

So straight off the bat my initial findings:

*1. Linting requiring react*
I've made an issue to split the lint rules into different configs, they're be exposed from the same package `@ts-engine/eslint-config` still, but they'll be separated. See #50 

*2. @ts-engine/eslint-config installation*
This is an interesting one and I'll need to look into it further. So the reason you got `Could not find module eslint-plugin-react` is probably the same reason I got `Could not find module eslint-plugin-prettier` when I installed `@ts-engine/cli` in your repository, the `eslint-plugin-prettier` was nested in `node_modules/@ts-engine/eslint-config/node_modules/eslint-plugin-prettier`. I've got no immediate idea why it would do this. It doesn't match the install pattern created when doing:

```
yarn global add @ts-engine/cli
ts-engine new-package --library --name my-lib
```

Try running that and checkout the node_modules and you'll see its not nested 🤔 See #52

*3. There is a Rollup compilation issue with class properties *
Not surprising this one slipped past me to be honest, I barely ever use classes. This isn't a big issue, its just a tweak to the ordering of rollup plugins. See #51

**Comment by LeeCheneler** [comment]
I'll try to get 1.6.0 out within the next few days with a fix for those 3 issues I've referenced 🤞

**Comment by LeeCheneler** [comment]
Turns out the reason that plugins were nested was because I had left prettier installed in your package.json, ts-engine provides prettier out the box so the versions conflicted.

**Comment by LeeCheneler** [comment]
I've opened the above PR as an example of how to migrate your library to use ts-engine 🙂 I hope that helps. I'll still aim to get #50 done in the next few days.

**Comment by kevinkhill** [comment]
> I'll try to get 1.6.0 out within the next few days with a fix for those 3 issues I've referenced crossed_fingers

No rush! That repo is just a toy :grin: 

Thank you though for all the quick replies and I'm happy to help contribute :smiley: 

I'm DrummerOfFenrir btw :wave:

**Comment by LeeCheneler** [comment]
Ah cool 😁✌️

**Comment by LeeCheneler** [comment]
@kevinkhill released `@ts-engine/cli@1.6.1` today which brings first class react support now with zero config. This splits the eslint react config into a new eslint preset only applied when you provide the `--react` argument. It also support react builds now by internally applying ts-engine's react abbel preset when providing the same flag.

```sh
ts-engine build --library --react
ts-engine build --node-app --react
ts-engine start --react
ts-engine lint --react
ts-engine test --react
```

**Comment by kevinkhill** [comment]
Awesome! Great work 👍🏻
